File: 6kyu/Rectangle into Squares/sqInRect.py
def sqInRect(lng, wdth):
    if lng == wdth:
        return None
    big = max(lng, wdth)
    small = min(lng, wdth)
    result = []
    while small > 0:
        result.append(small)
        small = min(big-small, small)
        big = max(big-small, small)
    return result


sqInRect(14, 4)

File: 6kyu/Rectangle into Squares/sqInRect.js
function sqInRect(lng, wdth) {
  //your code here
  if (lng === wdth) {
    return null;
  } else {
    let big = Math.max(lng, wdth);
    let small = Math.min(lng, wdth);
    let result = [];
    while (big > 0 && small > 0) {
      result.push(small);
      let temp = big - small;
      big = Math.max(temp, small);
      small = Math.min(temp, small);
    }
    return result
  }
}

